cocotb/regression.py

```diff
@@ -180,34 +185,41 @@ def _discover_tests() -> Iterable[Test]:
                 _logger.info(traceback.format_exc())
                 raise
 
-            if test_str:
-
+            if tests:
                 # Specific functions specified, don't auto-discover
-                for test_name in test_str.rsplit(','):
+                for test_name in tests:
                     try:
                         test = getattr(module, test_name)
                     except AttributeError:
-                        _logger.error("Requested test %s wasn't found in module %s", test_name, module_name)
-                        err = AttributeError("Test %s doesn't exist in %s" % (test_name, module_name))
-                        raise err from None  # discard nested traceback
-
-                    if not isinstance(test, Test):
-                        _logger.error("Requested %s from module %s isn't a cocotb.test decorated coroutine",
-                                      test_name, module_name)
-                        raise ImportError("Failed to find requested test %s" % test_name)
+                        not_found_tests.append(test_name)
+                    else:
+                        if not isinstance(test, Test):
+                            _logger.error("Requested %s from module %s isn't a cocotb.test decorated coroutine",
+                                          test_name, module_name)
+                            raise ImportError("Failed to find requested test %s" % test_name)
```

Comment:
Perhaps to minimize the diff:
```suggestion
                        continue

                    if not isinstance(test, Test):
                        _logger.error("Requested %s from module %s isn't a cocotb.test decorated coroutine",
                                      test_name, module_name)
                        raise ImportError("Failed to find requested test %s" % test_name)
```

Reply:
Changed in cb4ed6cd7a8e05c43ceda4fb305d6a20dbf4be1e.